provider/cluster/kube/client_exec.go

```diff
@@ -0,0 +1,207 @@
+package kube
+
+import (
+	"context"
+	"fmt"
+	"github.com/ovrclk/akash/provider/cluster"
+	ctypes "github.com/ovrclk/akash/provider/cluster/types"
+	mtypes "github.com/ovrclk/akash/x/market/types"
+	"io"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/remotecommand"
+	executil "k8s.io/client-go/util/exec"
+	"sort"
+	"strings"
+)
+
+// the type implementing the interface returned by the Exec command
+type execResult struct {
+	exitCode int
+}
+
+func (er execResult) ExitCode() int {
+	return er.exitCode
+}
+
+// a type to allow a slice of kubernetes pods to be sorted
+type sortablePods []corev1.Pod
+
+func (sp sortablePods) Len() int {
+	return len(sp)
+}
+
+func (sp sortablePods) Less(i, j int) bool {
+	return strings.Compare(sp[i].Name, sp[j].Name) == -1
+}
+
+func (sp sortablePods) Swap(i, j int) {
+	sp[i], sp[j] = sp[j], sp[i]
+}
+
+func (c *client) Exec(ctx context.Context, leaseID mtypes.LeaseID, serviceName string, podIndex uint, cmd []string, stdin io.Reader,
```

Comment:
can we use container name instead of pod index here? won't need to fancy with sorting pod names

actually, when deployment object is being created from the SDL, container name is set to same value as service name.
one of the examples below from SJC1
```
  template:
    metadata:
      creationTimestamp: null
      labels:
        akash.network: 'true'
        akash.network/manifest-service: mongo
        akash.network/namespace: 0lgvkaccl4gmi7lef591mnpu3tshmhlg0en1hsqv011di
    spec:
      containers:
        - name: mongo
          image: 'mongo:latest'
          args:
            - '--storageEngine=wiredTiger'
          ports:
            - containerPort: 27017
              protocol: TCP
          env:
```

Reply:
nvm, i'm wrong